pbf2json.go

```diff
@@ -29,30 +33,45 @@ func getSettings() Settings {
   // command line flags
   leveldbPath := flag.String("leveldb", "/tmp", "path to leveldb directory")
   tagList := flag.String("tags", "", "comma-separated list of valid tags, group AND conditions with a +")
-  batchSize := flag.Int("batch", 50000, "batch leveldb writes in batches of this size")
+  idList := flag.String("ids", "", "comma-separated list of valid ids")
+  batchSize := flag.Int("batch", 500, "batch leveldb writes in batches of this size")
 
   flag.Parse()
   args := flag.Args();
 
+  tagsConditions := make(map[string][]string)
+  idsConditions := make([]string,0,0)
+
   if len( args ) < 1 {
     log.Fatal("invalid args, you must specify a PBF file")
   }
 
-  // invalid tags
-  if( len(*tagList) < 1 ){
-    log.Fatal("Nothing to do, you must specify tags to match against")
-  }
-
-  // parse tag conditions
-  conditions := make(map[string][]string)
-  for _, group := range strings.Split(*tagList,",") {
-    conditions[group] = strings.Split(group,"+")
+  if len(*tagList) > 0 && len(*idList) > 0 {
+    log.Fatal("Please use either the -tags option or the -ids option, but not both at the same time")
+  } else if len(*tagList) < 1 && len(*idList) < 1 {
+    if len(*tagList) < 1 {
+      log.Fatal("Nothing to do, you must specify tags to match against")
+    } else { // len(*idList) < 1
+      log.Fatal("Nothing to do, you must specify id to match against")
+    }
+  } else if len(*tagList) > 0 {
+    // parse tag conditions
+    for _, group := range strings.Split(*tagList,",") {
+      tagsConditions[group] = strings.Split(group,"+")
+    }
+  } else {
+    // parse id conditions
+    for _, value := range strings.Split(*idList,",") {
+      idsConditions = append(idsConditions, value)
+      sort.Strings(idsConditions)
```

Comment:
minor performance improvement, could move this down one line and out of the loop?